src/evaluatorfrontend.cpp

```diff
@@ -67,18 +74,82 @@ void EvaluatorFrontend::FuncRunner() {
 		Plot();
 }
 
-//! @file
-// [runeval]
-void EvaluatorFrontend::RunEvaluator() {
-	while (!evaluator->IsFinished()) {
+int EvaluatorFrontend::EvaluatorFunc() {
+	if (!evaluator->IsFinished()) {
 		states.push_back(evaluator->GetNextNetworkState());
 		if (printStd) {
 			std::cout << states.back().PrintCsvRow();
 		}
+	} else {
+		return 0;
 	}
+
+	return 1;
+}
+
+//! @file
+// [runeval]
+void EvaluatorFrontend::RunEvaluator() {
+	while (EvaluatorFunc())
+		;
 }
 // [runeval]
 
+void EvaluatorFrontend::RunRTEvaluator() {
+	auto StartT = std::chrono::steady_clock::now();
+	auto EndT = std::chrono::steady_clock::now();
+	bool once = true;
+
+	while (!mPlot->RunPlot()) {
+		if (EvaluatorFunc()) {
+			EndT = std::chrono::steady_clock::now();
+			auto ETime =
+					std::chrono::duration_cast<std::chrono::milliseconds>(EndT - StartT);
+			auto UTime = std::chrono::milliseconds(UpdateRate);
+
+			if (ETime > UTime) {
+				StartT = std::chrono::steady_clock::now();
+
+				for (int i = 0; i < ToPlot.size(); i++) {
+					for (int j = LastInsertedCount; j < states.size(); j++) {
+						auto state = states[j];
+						ToPlot[i].Function.push_back((OpenRTP::Point){
+								(float)state.time, (float)state[ToPlot[i].Name]});
+					}
+				}
+
+				mPlot->UpdatePlot();
+				LastInsertedCount = states.size();
+			}
+		} else if (once) {
```

Comment:
Maybe rename `once` to `allStatesRendered`, or something that more clearly describes the intent behind the variable

Reply:
Okay, `allStatesRendered` is actually the opposite of what it's supposed to mean. You should probably add a `not` to the name, or just invert the value in the code.

Reply:
Is addRemainingStates better

Reply:
Maybe `remainingStatesNotAdded`? `addRemainingStates` is a verb, so it sounds like a function.

Reply:
It is kind of a function, and it's also the purpose but sure

Reply:
I see it as a bool. Are we talking about different things?

Reply:
No the thing that happens in the else if, but I will give it the better name